src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -370,6 +371,12 @@ private bool StartCore(ProcessStartInfo startInfo)
                 (userId, groupId, groups) = GetUserAndGroupIds(startInfo);
             }
 
+            // .NET applications don't echo characters unless there is a Console.Read operation.
+            // Unix applications expect the terminal to be in an echoing state by default.
+            // To support processes that interact with the terminal (e.g. 'vi'), we need to configure the
+            // terminal to echo. We keep this configuration as long as there are children possibly using the terminal.
+            bool usesTerminal = !startInfo.RedirectStandardInput && !startInfo.RedirectStandardOutput;
```

Comment:
Shouldn't this be || rather than &&?  And what about stderr?

Reply:
I wrote this check thinking, I want the terminal to echo when both standard in and standard out are connected to it.
Does that sound ok?
I can update the comments to reflect this better.

Reply:
added:

> // We consider the child to be interactively using the terminal when both stdin and stdout are connected.